### PR TITLE
Manifest updates

### PR DESCRIPTION
**Pull request by kaushikmitr**
This pull request includes several changes to Kubernetes manifests and configurations to update service names, deployment configurations, and role-based access controls.

### Service and Deployment Configuration Updates:

* [`examples/poc/manifests/llmservice.yaml`](diffhunk://#diff-1ce3dd8afd78d6ba5ca0e554eee539d09c58e93f43edf44ec86afe7bf4af2754R2-R11): Added a new `LLMServerPool` kind and updated the `models` section to include new model names and objectives. Updated the `poolRef` to use `vllm-llama2-7b-pool`. [[1]](diffhunk://#diff-1ce3dd8afd78d6ba5ca0e554eee539d09c58e93f43edf44ec86afe7bf4af2754R2-R11) [[2]](diffhunk://#diff-1ce3dd8afd78d6ba5ca0e554eee539d09c58e93f43edf44ec86afe7bf4af2754L10-R100)
* [`examples/poc/manifests/vllm/vllm-lora-deployment.yaml`](diffhunk://#diff-d8d58cad7c906e09d57425900fcd479ba45dd11e157672e3ce5844be08b45210R1-R29): Added a new `Service` for `vllm-llama2-7b-pool` and updated the `Deployment` configuration to use `vllm-llama2-7b-pool` with reduced replicas.
* [`examples/poc/manifests/vllm/vllm-lora-service.yaml`](diffhunk://#diff-1bb61c414fcca6522727dfc18ed98343d9382c6ad26127d877e2a319e527b98aL1-L12): Removed the old `Service` configuration for `vllm-lora`.

### Role-Based Access Control (RBAC) Updates:

* [`pkg/manifests/ext_proc.yaml`](diffhunk://#diff-b16d0243a556160f05d122667cec91ac324e751a07991e4f65c4e9015040ddc2R1-R31): Added new `ClusterRole` and `ClusterRoleBinding` for pod read access. Updated the deployment arguments to use `vllm-llama2-7b-pool` and added a verbosity level. [[1]](diffhunk://#diff-b16d0243a556160f05d122667cec91ac324e751a07991e4f65c4e9015040ddc2R1-R31) [[2]](diffhunk://#diff-b16d0243a556160f05d122667cec91ac324e751a07991e4f65c4e9015040ddc2L23-R61) [[3]](diffhunk://#diff-b16d0243a556160f05d122667cec91ac324e751a07991e4f65c4e9015040ddc2L63-R95)

### Gateway Configuration Updates:

* [`pkg/manifests/gateway.yaml`](diffhunk://#diff-4aca252918ba3acc926e6e739600eb9c846249765b8399fb55fc528d1006a74fL6-R8): Updated the gateway and gateway class names to `instance-gateway`. [[1]](diffhunk://#diff-4aca252918ba3acc926e6e739600eb9c846249765b8399fb55fc528d1006a74fL6-R8) [[2]](diffhunk://#diff-4aca252918ba3acc926e6e739600eb9c846249765b8399fb55fc528d1006a74fL20-R20) [[3]](diffhunk://#diff-4aca252918ba3acc926e6e739600eb9c846249765b8399fb55fc528d1006a74fL41-R41)
* [`pkg/manifests/patch_policy.yaml`](diffhunk://#diff-e62a193eb2a9985a8351fc22ecb41a825853025e188d928cfdee915312d803e8L10-R10): Updated the gateway name references in the patch policy to `instance-gateway`. [[1]](diffhunk://#diff-e62a193eb2a9985a8351fc22ecb41a825853025e188d928cfdee915312d803e8L10-R10) [[2]](diffhunk://#diff-e62a193eb2a9985a8351fc22ecb41a825853025e188d928cfdee915312d803e8L39-R39)

### CODE REVIEW

File: pkg/manifests/ext_proc.yaml [hunk 1]
```diff
@@ -20,13 +51,14 @@ spec:
...
         image: <BUILT-IMAGE>
         args:
-        # TODO(https://github.com/kubernetes-sigs/llm-instance-gateway/issues/12) Remove this once ext proc can dynamically reconcile on LLMServerPool.
-        - -pods
-        - "vllm-78665f78c4-h4kx4,vllm-78665f78c4-hnz84"
-        - -podIPs
-        - "10.24.11.6:8000,10.24.5.7:8000"
+        - -serverPoolName 
+        - "vllm-llama2-7b-pool"
+        - -v=3
```
Comment (liu-cong) [inline pkg/manifests/ext_proc.yaml:56]: let's be consistent with the other args:

```
- -v
- "3"
```

File: pkg/manifests/gateway.yaml [hunk 2]
```diff
@@ -3,9 +3,9 @@
 apiVersion: gateway.networking.k8s.io/v1
 kind: Gateway
 metadata:
-  name: <GATEWAY-NAME>
+  name: instance-gateway
```
Comment (liu-cong) [inline pkg/manifests/gateway.yaml:6]: Perhaps call it inference-gateway?

File: examples/poc/manifests/vllm/vllm-lora-deployment.yaml [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+metadata:
+  name: vllm-llama2-7b-pool
+spec:
+  selector:
+    app: vllm-llama2-7b-pool
+  ports:
+  - protocol: TCP
+    port: 8000
+    targetPort: 8000
+  type: LoadBalancer
```
Comment (liu-cong) [inline examples/poc/manifests/vllm/vllm-lora-deployment.yaml:12]: Let's change it to ClusterIP.

**Review by liu-cong — COMMENTED** [review]
/lgtm

Thanks!

**Comment by ahg-g** [comment]
@kaushikmitr can you address @liu-cong comments pls on this one? all are minor.

**Comment by kaushikmitr** [comment]
done

**Pull request merged**